src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -25,184 +27,233 @@
 #include "msquichelper.h"
 #include "msquic.hpp"
 
+#define MUST_SUCCEED(X) CXPLAT_FRE_ASSERT(QUIC_SUCCEEDED(X))
+
 const MsQuicApi* MsQuic;
-uint64_t MagicCid = 0x989898989898989ull;
+static const char* Alpn = "fuzz";
+static uint32_t Version = QUIC_VERSION_1;
+const char* Sni = "localhost";
+const StrBuffer InitialSalt("38762cf7f55934b34d179ae6a4c80cadccbb7f0a");
 const QUIC_HKDF_LABELS HkdfLabels = { "quic key", "quic iv", "quic hp", "quic ku" };
+const uint64_t MagicCid = 0x989898989898989ull;
+const uint16_t MinInitialDatagramLength = 1200;
 uint64_t RunTimeMs = 60000;
 CxPlatEvent RecvPacketEvent(true);
 QUIC_RX_PACKET* PacketQueue;
 QUIC_RX_PACKET** PacketQueueTail = &PacketQueue;
 CxPlatLock PacketQueueLock;
 uint64_t CurrSrcCid = 0;
 
+struct FuzzerStats {
+    int64_t RecvDatagrams = 0;
+    int64_t RecvInitialPackets = 0;
+    int64_t RecvHandshakePackets = 0;
+    int64_t Recv1RttPackets = 0;
+
+    int64_t SendDatagrams = 0;
+    int64_t SendInitialPackets = 0;
+    int64_t SendHandshakePackets = 0;
+    int64_t Send1RttPackets = 0;
+
+    void Print() {
+        printf("\n");
+        printf("Send:\n");
+        printf("  Datagrams:         %lld\n", (long long)SendDatagrams);
```

Comment:
Probably time to get into the habit of using volatile accessors.

Reply:
I'm on my way to making this all single threaded.